test/functional/legacy/065_float_and_logic_operators_spec.lua

```diff
@@ -0,0 +1,182 @@
+-- Test for floating point and logical operators.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('floating point and logical operators', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([=[
+      Results of test65:]=])
+
+    source([[
+      $put =printf('%f', 123.456)
+      $put =printf('%e', 123.456)
+      $put =printf('%g', 123.456)]])
+    -- Check we don't crash on division by zero.
+    source([[
+      echo 1.0 / 0.0
+      $put ='+='
+      let v = 1.234
+      let v += 6.543
+      $put =printf('%g', v)
+      let v = 1.234
+      let v += 5
+      $put =printf('%g', v)
+      let a = 5
+      let a += 3.333
+      $put =string(a)
+      $put ='=='
+      let v = 1.234
+      $put =v == 1.234
+      $put =v == 1.2341
+      $put ='add-subtract'
+      $put =printf('%g', 4 + 1.234)
+      $put =printf('%g', 1.234 - 8)
+      $put ='mult-div'
+      $put =printf('%g', 4 * 1.234)
+      $put =printf('%g', 4.0 / 1234)
+      $put ='dict'
+      $put =string({'x': 1.234, 'y': -2.0e20})
+      $put ='list'
+      $put =string([-123.4, 2.0e-20])
+      $put ='abs'
+      $put =printf('%d', abs(1456))
+      $put =printf('%d', abs(-4))
+    ]])
+    -- The test seems to fail if this execute call is put inside the above or
+    -- the below source() call.
+    execute([[$put =printf('%d', abs([1, 2, 3]))]])
```

Comment:
Wow, I'm very sorry, I simply did not see your comment in the code. I must be very tired.

Anyways then I think you can leave it as it is but please change the comment to explain *why* the line must not be inside `source`. The reason is that the expression is actually invalid and throws an exception (`Using an X as a Number` ...). The legacy test runner simply ignores the error and continues on, whereas our `source` ends the session when it encounters an exception. `execute` doesn't have this problem because it feeds keys directly to nvim I believe (I didn't dig any further than that.)